src/Disks/ObjectStorages/VFSLogItem.cpp

```diff
@@ -0,0 +1,160 @@
+#include "VFSLogItem.h"
+#include "Common/logger_useful.h"
+#include "IO/ReadBufferFromMemory.h"
+#include "IO/ReadHelpers.h"
+#include "IO/WriteHelpers.h"
+
+using ItemPair = DB::VFSLogItemStorage::value_type;
+template <>
+struct fmt::formatter<ItemPair>
+{
+    constexpr auto parse(auto & ctx) { return ctx.begin(); }
+    constexpr auto format(const ItemPair & item, auto & ctx) { return fmt::format_to(ctx.out(), "{} {}", item.first, item.second); }
+};
+
+namespace DB
+{
+VFSLogItem VFSLogItem::parse(std::string_view str)
+{
+    if (str.empty() || str == "\n")
+        return VFSLogItem{};
+    VFSLogItem out;
+    ReadBufferFromMemory buf{std::move(str)};
+    ItemPair pair;
+    while (!buf.eof())
+    {
+        readStringUntilWhitespace(pair.first, buf);
+        checkChar(' ', buf);
+        readIntTextUnsafe(pair.second, buf);
+        checkChar('\n', buf);
+        out.emplace(std::exchange(pair, {}));
```

Comment:
What is the idea behind out.emplace(std::exchange(pair, {}))  ,
why not simply out.emplace(std::move(pair))
?